src/hooks/screen/useMessageHook.ts

```diff
@@ -75,11 +76,29 @@ function useMessageHook(): UseMessageHook {
     }
   };
 
+  const onOpenMediaPreview = (medias, index, navigation) => {
+    if (medias.find((media) => media.type === 'video')) {
+      navigation.push('VideoViewer', {
+        title: 'Video',
+        url: medias[index].video_path
+      });
+    } else {
+      navigation.push('ImageViewer', {
+        title: 'Photo',
+        index,
+        images: medias
+          .filter((media) => media.type === 'photo')
+          .map((media) => ({url: media.asset_url}))
+      });
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `onOpenMediaPreview` function assumes that if any media in the `medias` array is a video, then all are videos. This could lead to incorrect behavior if there are both photos and videos in the `medias` array. Consider iterating over each media item and handling them based on their type.

```diff
-    if (medias.find((media) => media.type === 'video')) {
-      navigation.push('VideoViewer', {
-        title: 'Video',
-        url: medias[index].video_path
-      });
-    } else {
-      navigation.push('ImageViewer', {
-        title: 'Photo',
-        index,
-        images: medias
-          .filter((media) => media.type === 'photo')
-          .map((media) => ({url: media.asset_url}))
-      });
-    }
+    medias.forEach((media, index) => {
+      if (media.type === 'video') {
+        navigation.push('VideoViewer', {
+          title: 'Video',
+          url: media.video_path
+        });
+      } else if (media.type === 'photo') {
+        navigation.push('ImageViewer', {
+          title: 'Photo',
+          index,
+          images: medias
+            .filter((media) => media.type === 'photo')
+            .map((media) => ({url: media.asset_url}))
+        });
+      }
+    });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->